llmfoundry/models/layers/blocks.py

```diff
@@ -52,23 +52,18 @@ def __init__(
         super().__init__()
 
         norm_class = NORM_CLASS_REGISTRY[norm_type.lower()]
-        assert isinstance(attn_config['attn_type'], str)
-        attn_class = ATTN_CLASS_REGISTRY[attn_config['attn_type']]
+
+        attn_type = attn_config_mutable.pop('attn_type')
```

Comment:
I don't love the mutable config thing. What do you think about the option where inside this function you just make a copy of the attn_config with the extraneous args removed? Then the caller of the block creation does not need to do any popping, and this function doesn't have side effects on the original attn config.

Reply:
yea I think that might be cleaner. I'll change the implementation accordingly.